lbjava/src/main/java/edu/illinois/cs/cogcomp/lbjava/neuralnet/NeuralNetwork.java

```diff
@@ -0,0 +1,200 @@
+            else if (args[i].equals("-help")) {
+                System.out.println("-t the number of threads to deploy.\n-l the learning rate.\n-m momentum.\n-e number of epochs.\n-h hidden layer size.");
+                System.exit(0);
+            } else
+                System.out.println("Unexpected argument : "+args[i]);
+        }
+    }
+    /**
+     * Given the number of input layers and outputs, and the sizes of all layers, 
+     * set up an untrained neural net.
+     * @param layerSizes the number of neurons in each layer, also corresponds to the number of outputs of that layer.
+     * @param learningRate the learning rage.
+     * @param momentum the momentum.
+     */
+    NeuralNetwork(int[] layerSizes) {
+        layers = new Layer[layerSizes.length-1];
+
+        // each layer has a number of inputs defined by the outputs of the previous layer, or
+        // the number inputs passed in, outputs is the number of neurons in the layer since each 
+        // neuron produces one output.
+        Random r = new Random (1234);
+        for (int i = 0; i < layerSizes.length-1; i++) {
+            this.layers[i] = new Layer(layerSizes[i], layerSizes[i+1], r);
+        }
+    }
+    
+    /**
+     * @see edu.illinois.cs.cogcomp.lbjava.neuralnet.Activator#activateLayers(float[], edu.illinois.cs.cogcomp.lbjava.neuralnet.Layer[])
+     */
+    @Override
+    public float[] prediction(float[] inputs) {
+        // set up our counts.
+        int layerCount = layers.length;
+        
+        // storage for each output of each layer, and the error computed for each activation.
+        float [][] activations = new float[layerCount][];
+        
+        // This array contains inputs from previous layer
+        float [] currentinputs = inputs;
+        for (int i = 0 ; i < layerCount ; i++) {
+            
+            // compute the activations for this layer.
+            Layer layer = layers[i];
+            activations[i] = layer.activate(currentinputs);
+            currentinputs = activations[i];
+        }
+        return activations[layerCount-1];
+    }
+
+    /** 
+     * Train up the NN model given training data, a learner algorith,
+     * and convergence criteria.
+     * @param inputs the input data.
+     * @param outputs the desired output.
+     * @param learner the learning algorithm.
+     * @param epochs number of iterations to run.
+     * @param converge the convergence criteria.
+     */
+    public void train(float[][] inputs, float[][]outputs, NNTrainingInterface learner, int epochs) {
+        if (inputs.length != outputs.length)
+            throw new RuntimeException("There must be the same number of input data records and output data records to train.");
+        learner.train(inputs, outputs, epochs);
+    }
+    
+    /**
+     * Test will try learning an XOR model.
+     * @param args
+     * @throws IOException 
+     */
+    public static void main(String[] args) throws IOException {
```

Comment:
Shouldn't we move this example next to other lbjava examples, under `lbjava-examples`?

Reply:
@danyaljj I could put this in examples, but most of the stuff in there is LBJava. This is really something I used to test the NN locally, without LBJava. I could move this to examples, or I could convert to LBJava (this would take some time), or I could just leave it where it is. I don't think it hurts to just leave it.

Reply:
I think the library code purely the code that's being used for creating/training models and it doesn't contain any examples. I'd prefer if we move the example (especially `NNBrownClassifier.lbj` to the examples module).

Reply:
Ah, sorry, I was misunderstanding, I thought you wanted this main method moved. Yes, you are right, I will move that.

Reply:
I have moved the LBJava file, and all files only used by this example into the lbjava-examples project.